### PR TITLE
tests: add version structure tests

### PR DESCRIPTION
**Pull request by meenal06**
Signed-off-by: Meenal Trivedi <meenaltrivedi6102@gmail.com>

### CODE REVIEW

**Comment by meenal06** [comment]
Hi @callahad @Kegsay , please review the following PR. I didn't find a method in our utils to check if the key-type is an array or not, hence I ended up using `match.JSONArrayEach` with an empty function block, as the function has a pre-check for key to be an array type.

File: tests/apidoc_version_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: Version responds 200 OK with valid structure
+	t.Run("Version responds 200 OK with valid structure", func(t *testing.T) {
+		res := unauthedClient.MustDo(t, "GET", []string{"_matrix", "client", "versions"}, json.RawMessage(`{}`))
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			JSON: []match.JSON{
+				match.JSONKeyPresent("versions"),
+				match.JSONArrayEach("versions", func(val gjson.Result) error {
+					return nil
```
Comment (Kegsay) [inline tests/apidoc_version_test.go:26]: You can do more here. The original tests don't assert anything on the elements, but we can at least make sure they are strings by doing:
```go
match.JSONArrayEach("versions", func(val gjson.Result) error {
    if val.Type != gjson.String {
        return fmt.Errorf("value isn't string: %s", val.Raw)
    }
    return nil
}
```

This uses the `gjson` library which we use a lot in Complement - see the functions/fields you can use [here](https://pkg.go.dev/github.com/tidwall/gjson#Result).

**Comment by meenal06** [comment]
Hi @Kegsay I have done the requested changes, please review 😄 !

File: tests/apidoc_version_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestVersionStructure(t *testing.T) {
+	deployment := Deploy(t, "test_version", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: Version responds 200 OK with valid structure
+	t.Run("Version responds 200 OK with valid structure", func(t *testing.T) {
+		res := unauthedClient.MustDo(t, "GET", []string{"_matrix", "client", "versions"}, json.RawMessage(`{}`))
```
Comment (Kegsay) [inline tests/apidoc_version_test.go:21]: GET requests will never have a body, so please replace `json.RawMessage(`{}`)` with `nil`.

Comment (meenal06) [reply tests/apidoc_version_test.go]: Done!

**Comment by meenal06** [comment]
@Kegsay I did the requested change, please do take a look again!

**Review by Kegsay — APPROVED** [review]

**Pull request merged**